extension/public/reddit.js

```diff
@@ -123,21 +134,40 @@ var Reddit = {
   // retrive the posts iamge link
   getDOMImageLink: function (_post) {
     classSelector = "._2_tDEnGMLxpM6uOa2kaDB3";
-    var url = _post.querySelector(classSelector).getAttribute("src");
-    console.log("image url: " + url); // DEBUG
+    var url = _post.querySelector(classSelector);
+    if(url != null){
+      url = url.getAttribute("src");
+      console.log("image url: " + url); // DEBUG
+      return url;
+    }else{
+      return "n/a";
+    }
   },
 
   getDOMTitle: function (_post) {
     classSelector = "._eYtD2XCVieq6emjKBH3m";
-    var title = _post.querySelector(classSelector).textContent;
-    console.log("post title: " + title); // DEBUG
+    var title = _post.querySelector(classSelector);
+    if(title != null){
+      title = title.textContent.toLowerCase();
+      console.log("post title: " + title); // DEBUG
+      return title;
+    }else{
+      return "n/a";
```

Comment:
See previous about error handling, use this for reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw